src/utils/mod.rs

```diff
@@ -0,0 +1,233 @@
+// SPDX-FileCopyrightText: 2020 Serokell <https://serokell.io/>
+//
+// SPDX-License-Identifier: MPL-2.0
+
+use std::borrow::Cow;
+use std::path::PathBuf;
+
+use merge::Merge;
+
+#[macro_export]
+macro_rules! good_panic {
+    ($($tts:tt)*) => {{
+        error!($($tts)*);
+        std::process::exit(1);
+    }}
+}
+
+pub mod data;
+pub mod deploy;
+pub mod push;
+
+pub struct CmdOverrides {
+    pub ssh_user: Option<String>,
+    pub profile_user: Option<String>,
+    pub ssh_opts: Option<String>,
+    pub fast_connection: Option<bool>,
+    pub auto_rollback: Option<bool>,
+    pub hostname: Option<String>,
+    pub magic_rollback: Option<bool>,
+    pub temp_path: Option<String>,
+    pub confirm_timeout: Option<u16>,
+}
+
+#[derive(PartialEq, Debug)]
+pub struct DeployFlake<'a> {
+    pub repo: &'a str,
+    pub node: Option<&'a str>,
+    pub profile: Option<&'a str>,
+}
+
+pub fn parse_flake(flake: &str) -> DeployFlake {
```

Comment:
Does Nix parse flakes in the same way? If it does, how does it handle paths with `#` ?

Reply:
I'm not sure what you mean, the flake URL given to `nix build` and the one given to `deploy` will already be very different as stuff like `.deploy` `.nodes`, .`profiles`, and `.path` are left out

Reply:
Oh I think I get what you mean now, I didn't add any form of escaping at all, I probably should

Reply:
Well, it looks like Nix doesn't let you have `#` in your path either, and has no escaping at all, so I guess there's no point fixing that if `nix` will fail when it encounters it anyway, and it would just create more inconsistencies